aya/src/maps/mod.rs

```diff
@@ -197,6 +191,10 @@ pub enum MapError {
         #[source]
         error: PinError,
     },
+
+    /// The map is not of the expected type.
+    #[error("unexpected map type")]
+    UnexpectedMapType,
```

Comment:
I don't think we need this? See `InvalidMapType`

Reply:
I made a new one here specifically to be used when an issue is encounter with casting to concrete Aya::Map types since an `Aya:Map` doesn't have an int value like an  `generated::bpf_map_type` does 

`InvalidMapType` IMO is used for when we encounter an invalid map type at load time

Reply:
Happy to change though if you think otherwise :)

Reply:
Well, a new variant is new API, and imagine being the caller and having to handle  both `UnexpectedMapType` and `invalidMapType` - doesn't seem like a great API. So I think the solution is to use the same variant for both.

`InvalidMapType` is now used to report unsupported map types at load time. It's definitely useful to know which map type is failing, at least to us if people report an issue about it.

Here we don't have a `bpf_map_type` but we could create a utility that maps `Map` variants to `bpf_map_type`? And then we could use `InvalidMapType` here too

Reply:
Done!